runtime/js/10_dispatch_buffer.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+
+  function assert(cond) {
+    if (!cond) {
+      throw Error("assert");
+    }
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////// General async handling //////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  // General Async response handling
+  let nextRequestId = 1;
+  const promiseTable = {};
+
+  function opAsync(opName, opRequestBuilder, opResultParser) {
+    // Make sure requests of this type are handled by the asyncHandler
+    // The asyncHandler's role is to call the "promiseTable[requestId]" function
+    core.setAsyncHandlerByName(opName, (bufUi8, _) => {
```

Comment:
Also wondering if this should be called only once, otherwise a new callback is set with each call to the op

Reply:
I think this should be called once. The current jsonOp implementation also does this wrongly:
https://github.com/denoland/deno/blob/bd961c3bc305e90e760394d2f424c5360a67543b/core/core.js#L214-L217
I do fix this for both bufferOps and jsonOps in #9457.